api/user_manager.py

```diff
@@ -0,0 +1,143 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2023 Collabora Limited
+# Author: Jeny Sadadia <jeny.sadadia@collabora.com>
+
+"""User Manager"""
```

Comment:
It might be a bit better to explain what this module is about.  Maybe docstrings and code clarity in general is something we can take care as a follow-up.